app/src/main/java/org/jellyfin/androidtv/ui/playback/nextup/NextUpViewModel.kt

```diff
@@ -52,22 +56,34 @@ class NextUpViewModel(
 	}
 
 	private suspend fun loadItemData(id: String) = withContext(Dispatchers.IO) {
+		val nextUpThumbnailEnabled = userPreferences[UserPreferences.nextUpThumbnailEnabled]
 		val item = apiClient.getItem(id) ?: return@withContext null
 
-		val thumbnail = apiClient.GetImageUrl(item, ImageOptions())
+		val thumbnail = if (nextUpThumbnailEnabled) apiClient.GetImageUrl(item, ImageOptions()) else null
 		val logo = apiClient.GetLogoImageUrl(item, ImageOptions())
 
-		val title = if (item.indexNumber != null && item.name != null)
-			"${item.indexNumber}. ${item.name}"
-		else if (item.name != null)
-			item.name
-		else ""
+		val title = when {
+			(item.parentIndexNumber != null && item.parentIndexNumber != 0 && item.indexNumber != null && item.indexNumberEnd != null && item.name != null) ->
+				"${context.getString(R.string.lbl_season_number, item.parentIndexNumber)}:${context.getString(R.string.lbl_episode_number, item.indexNumber)}-${item.indexNumberEnd} - ${item.name}"
+			(item.parentIndexNumber != null && item.parentIndexNumber != 0 && item.indexNumber != null && item.name != null) ->
+				"${context.getString(R.string.lbl_season_number, item.parentIndexNumber)}:${context.getString(R.string.lbl_episode_number, item.indexNumber)} - ${item.name}"
+			(item.parentIndexNumber != null && item.parentIndexNumber == 0 && item.name != null) ->
+				"${context.getString(R.string.lbl_special)} - ${item.name}"
+			(item.parentIndexNumber != null && item.name != null) ->
+				"${context.getString(R.string.lbl_season_number, item.parentIndexNumber)} - ${item.name}"
+			(item.indexNumber != null && item.indexNumberEnd != null && item.name != null) ->
+				"${context.getString(R.string.lbl_episode_number, item.indexNumber)}-${item.indexNumberEnd} - ${item.name}"
+			(item.indexNumber != null && item.name != null) ->
+				"${context.getString(R.string.lbl_episode_number, item.indexNumber)} - ${item.name}"
+			(item.name != null) ->
+				item.name
+			else -> ""
+		}
```

Comment:
This is really difficult to read/understand. You should find another way to create the strings. (and do we really need this many?!)

Reply:
The string is now built as 3 pieces joined together—should be a lot easier to read than this mess